engine/sim/cooldown.hpp

```diff
@@ -121,3 +122,29 @@ struct cooldown_t
 private:
   void adjust_remaining_duration( double delta ); // Modify the remaining duration of an ongoing cooldown.
 };
+
+struct target_specific_cooldown_t
+{
+  player_t* const player;
+  cooldown_t* const base_cooldown;
```

Comment:
I think just storing the duration and name here would be better than actually referencing the regular cooldown. It's possible someone could want to use this without creating a base cooldown.

Reply:
My thought process here for the time being is that I wasn't totally sure if it worked with RPPM objects yet (it is assigned to that one RPPM spell but I did not have time to test if it actually functions) and I figured we may need additional aspects of the cooldown data. But just storing the duration could be sufficient for now.